src/main/scala/cognite/spark/v1/FlexibleDataModelRelationUtils.scala

```diff
@@ -571,122 +600,124 @@ object FlexibleDataModelRelationUtils {
-            val doubleSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            tryAsDoubleSeq(doubleSeq, propertyName)
-              .map(InstancePropertyValue.Float64List)
-          case p @ PrimitiveProperty(PrimitivePropType.Int32, _) if p.isList =>
-            val intSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            tryAsIntSeq(intSeq, propertyName)
-              .map(InstancePropertyValue.Int32List)
-          case p @ PrimitiveProperty(PrimitivePropType.Int64, _) if p.isList =>
-            val longSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            tryAsLongSeq(longSeq, propertyName)
-              .map(InstancePropertyValue.Int64List)
-          case p @ PrimitiveProperty(PrimitivePropType.Timestamp, _) if p.isList =>
-            val tsSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            tryAsTimestamps(tsSeq, propertyName).map(InstancePropertyValue.TimestampList)
-          case p @ PrimitiveProperty(PrimitivePropType.Date, _) if p.isList =>
-            val dateSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            tryAsDates(dateSeq, propertyName).map(InstancePropertyValue.DateList)
-          case p @ PrimitiveProperty(PrimitivePropType.Json, _) if p.isList =>
-            val strSeq = Try(row.getSeq[String](i)).getOrElse(row.getAs[Array[String]](i).toSeq)
-            skipNulls(strSeq).toVector
-              .traverse(io.circe.parser.parse)
-              .map(InstancePropertyValue.ObjectList.apply)
-              .leftMap(e =>
-                new CdfSparkException(
-                  s"Error parsing value of field '$propertyName' as a list of json objects: ${e.getMessage}"))
-          case p: TimeSeriesReference if p.isList =>
-            val strSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            Try(InstancePropertyValue.TimeSeriesReferenceList(skipNulls(strSeq).map(String.valueOf))).toEither
-          case p: FileReference if p.isList =>
-            val strSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            Try(InstancePropertyValue.FileReferenceList(skipNulls(strSeq).map(String.valueOf))).toEither
-          case p: SequenceReference if p.isList =>
-            val strSeq = Try(row.getSeq[Any](i)).getOrElse(row.getAs[Array[Any]](i).toSeq)
-            Try(InstancePropertyValue.SequenceReferenceList(skipNulls(strSeq).map(String.valueOf))).toEither
-          case t => Left(new CdfSparkException(s"Unhandled list type: ${t.toString}"))
+    Try(schema.fieldIndex(propertyName)) match {
+      case Failure(_) => Right(RowInstancePropertyValue.NotPresent)
+      case Success(i) =>
+        if (row.isNullAt(i)) {
+          if (nullable) {
+            Right(RowInstancePropertyValue.IsNull)
+          } else {
+            Left(new CdfSparkException(s"'$propertyName' cannot be null"))
```

Comment:
Hmm, I wonder if this exception should be thrown only when `ignoreNullFields = false`. If `ignoreNullFields = true`, null fields should behave the same as fields not present, I think? Which should be valid.

This is different from what used to be the behavior, which assumed `replace = true`, in which case both omitting and setting to null would be invalid on a non-nullable property.

Same for the other cases below. Thoughts?